### DISCUSSION TITLE
paperless-ng api pdf as string

### DISCUSSION THREAD

**Discussion by toupou** [jonaswinkler/paperless-ng#1457] [Support]
Dear all, 

I´m trying to use Node-Red to get that pdf out of paperless-ng and getting following response:

Header:
`{"date":"Sun, 28 Nov 2021 18:29:43 GMT","server":"uvicorn","content-type":"application/pdf","content-disposition":"attachment; filename=\"2021-11-26 hello world.pdf\"","vary":"Accept, Accept-Language, Origin, Cookie","allow":"GET, HEAD, OPTIONS","x-frame-options":"SAMEORIGIN","x-api-version":"2","x-version":"1.5.0","content-length":"17270","content-language":"en-us","x-content-type-options":"nosniff","referrer-policy":"same-origin","connection":"close","x-node-red-request-node":"40e38a65"}`

Payload (string):
`%PDF-1.7%����1 0 obj<< /Type /ObjStm /Length 704 /Filter /FlateDecode /N 11 /First 67 >>streamx�uTmk�0��_qߖR[r�6J ��.�4!����RQ�2�Ϳߝ�Y��`Kw���sϙ��1��'!��"�b�� ��!q`Rz�W��o����`��[��j�������P�o%���t�m{�x3�,��c6i���0Xt��6�A��P���JXm�LX���9c�G�s�׾��ޏ@L^O��a��`����jU���r��Yew0�!à��p�Q�o����6W\�E�|�"�ն�X���k�:�33mi���]K�"�����{���Z(��ԼbQ��'86LҎ��{>ת1m�#^V�zS�\����i��|�j<vM�e1�Gtx[�j�Lo%R��0�ͦ�QعW6]9lo�-���(�s���T��R�Ғ�I���-s#u�u�m�g�RZ_�G��G�/�R�j!|Qⓐ�oJ��c���M�!�ؓ�\�'-Q�:��t���N��zG���9JV�r�3�T�4�ɣ���a/V��ˢ��ZW��s�k\�� �=l�<8�,�\)�Դs���&-6��)*p��v�g8a}��E�s�Z�y�95X����p�o��~�-D�[d/{����]%����A����=��6�1r���<	���%�Qg���!�=�-,k�j�Π����H�{�����4ϊi��P���C�aeb�endstreamendobj13 0 obj<< /Subtype /XML /Type /Metadata /Length 1340 >>stream<?xpacket begin="﻿" id="W5M0MpCehiHzreSzNTczkc9d"?><x:xmpmeta xmlns:x="adobe:ns:meta/" x:xmptk="XMP toolkit 2.9.1-13, framework 1.6"><rdf:RDF xml...`

If I then try to save that as a pdf or forward via mail the pdf is corrupt and not able to be opened. 

Does anyone have a clue and point me in the right direction how I need to encode/decode this?

Thanks
Philipp

**Comment by ghost** [comment]
This is my function-node:
`filename = msg.filename.split("/");
msg.path = "/tmp";
filename = filename.pop();

msg.headers = {
    "Content-Type": "multipart/form-data; boundary=------------------------d74496d66958873e"
}


msg.payload = '--------------------------d74496d66958873e\r\n'+
'Content-Disposition: form-data; name="title"\r\n'+
'Content-Type: application/json\r\n'+
'{"title": '+filename+'}\r\n'+
'true\r\n'+
'--------------------------d74496d66958873e\r\n'+
'Content-Disposition: form-data; name="correspondent"\r\n'+
'Content-Type: application/json\r\n'+
'{"correspondent": '+msg.device+'}\r\n'+
'true\r\n'+
'--------------------------d74496d66958873e\r\n'+
'Content-Disposition: form-data; name="document"; filename="'+msg.filename+'"\r\n'+
'Content-Type: application/octet-stream\r\n'+
'\r\n'+
msg.payload+'\r\n'+
'--------------------------d74496d66958873e--\r\n';


return msg;`

Please try and give feedback.
